palaverapi/tests/test_views.py

```diff
@@ -66,3 +67,113 @@ def enqueue(*args):
 
         token.delete_instance()
         device.delete_instance()
+
+
+class AuthorisationListViewTests(unittest.TestCase):
+    def setUp(self):
+        self.client = TestClient(router)
+
+        self.device = Device.create(apns_token='ec1752bd70320e4763f7165d73e2636cca9e25cf')
+        self.token = Token.create(device=self.device, token='e4763f7165d73e2636cca9e', scope=Token.ALL_SCOPE)
+
+    def tearDown(self):
+        self.device.delete_instance()
+        self.token.delete_instance()
```

Comment:
The tests on CircleCI fail currently with error: ```IntegrityError: update or delete on table "device" violates foreign key constraint "token_device_id_fkey" on table "token"```. I think swapping these lines (delete tokens first) will fix the issue, as well as the failed transaction errors down the line.